spec/system/support_interface/add_course_to_submitted_application_spec.rb

```diff
@@ -32,6 +32,14 @@
     when_i_select_a_course
     and_i_click_add_course_to_application
     then_i_should_see_the_application_with_the_course_added
+
+    when_there_are_two_more_courses_added
+    and_i_visit_the_application_page
+    then_i_should_not_be_able_to_add_further_courses
+
+    when_one_course_is_withdrawn
+    and_i_visit_the_application_page
+    then_i_should_be_able_to_add_further_courses
```

Comment:
Consider moving these tests to the AddCourseComponent, where we should be verifying all possible scenarios.